app/models/ability.rb

```diff
@@ -0,0 +1,18 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    if user.admin == true
+      can :manage, :all
+    else
+      can :destroy, Reservation do |reservation|
+        reservation.user == user
+      end
+
+      can :update, Reservation do |reservation|
+        reservation.user == user
+      end
+      can :create, Reservation
+    end
```

Comment:
It would be better to include that the regular user, )not admin)
- can only read their user data
- can only read their reservation data
- can read all fitness activity and available date data

i.e. 
```
can :read, Users, id: user.id
can :read, Reservation, user_id: user.id
can :read, FitnessActivity
can :read, AvailableDate
```